ERCS/erc-7555.md

```diff
@@ -0,0 +1,211 @@
+
+Not being able to correctly identify existing accounts owned by a user will lead to account fragmentation. The fragmentation, as described early, exists because applications will identify them as a new user, and not one whom may already have an account. Leading to a single user having many unassociated accounts, with assets scattered amongst them, and no way to unify them.
+
+This standard aims to achieve:
+1. Standard way for applications to request a users signing address.
+2. Standard way for applications to provide single sign-on (SSO) functionality for alternative signing methods.
+3. Standard way for applications to disclose smart accounts that have been created through their own service.
+
+This standard **does not** aim to achieve:
+1. How a user can sign messages across domains.
+2. How a provider generates a keypair for a user.
+3. How an application handles the user interface logic.
+
+
+## Specification
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Definitions
+- **Smart account** - An ERC-4337 compliant smart contract account that has a modular architecture.
+- **Domain** - A string of text acting as an identification to a server or wesbite (eg: `ethereum.org` or `ABCDE12345.com.example.app`).
+- **EOA** - Accounts that are controlled by a single private key.
+- **Provider** - A third party service provider that is able to authenticate a user and produce a keypair for the user.
+
+### Redirects
+An application looking to authenticate a user must navigate the user to a given provider's URI based on the `URI Request Syntax`. The application must implement a valid redirect URI for the callback in order to receive a valid response.
+
+#### Available Routes
+- `/auth/`: The route used to authenticate a user, and request credentials.
+- `/sendTransaction/`: The route used to send a transaction payload for a user to sign. This is more of a convenient method to allow applications to do both authentication, and plugin registration within a single redirect, instead of requiring the user to perform two redirects.
+
+### Schema
+The `smart_account_address` should be returned in the CAIP-10 format.
+
+#### Auth Route
+##### Request Schema
+```=
+ parameters:
+    - in: query
+      name: redirect_uri
+      schema:
+        type: string
+      description: The uri that the provider should redirect back to.
+   - in: query
+      name: chain_id
+      schema:
+        type: string
+      description: The chain_id of a given network.
+```
+##### Response Schema
+```=
+ parameters:
+  - in: query
+      name: smart_account_address
+      schema:
+        type: string
+      description: The on-chain address for a given smart account, formatted using CAIP-10
+```
+
+##### Request Syntax
+```=
+https://<PROVIDER_URI>/auth/?
```

Comment:
sorry if this is addressed elsewhere but where does the dapp get the `PROVIDER_URI` from?

Reply:
That's undefined per this spec, it would be sourced from something external such as walletconnect directly applying it OR [CAIP-275](https://chainagnostic.org/CAIPs/caip-275)